internal/service/network/network.go

```diff
@@ -25,20 +25,25 @@ var (
 	errNetworkIDNotFound    = errors.New("network ID not found")
 )
 
+type mutexCounter struct {
+	mutex      *sync.Mutex
+	activeReqs int // activeReqs will track concurrent active requests for the same network
```

Comment:
does using a atomic counters simplify your code?

Reply:
Using an atomic won't simplify the code. I will still need to have the mutex and hence the mutexCounter struct,  since the mutex is used by the network create/remove to block concurrent execution